src/line_parser2.rs

```diff
@@ -0,0 +1,238 @@
+use crate::{BeatNumber, Line, LineNote, Result};
+use combine::{
+    choice, many1, optional,
+    parser::char::{digit, space, spaces},
+    sep_by, token, Parser,
+};
+
+#[derive(Clone, Copy, Debug)]
+enum Letter {
+    A,
+    B,
+    C,
+    D,
+    E,
+    F,
+    G,
+}
+
+#[derive(Clone, Copy, Debug)]
+enum Modifier {
+    Flat,
+    Natural,
+}
+
+#[derive(Clone, Copy, Debug)]
+struct Note {
+    letter: Letter,
+    modifier: Modifier,
+    octave: i8,
+    duration: u32,
+}
```

Comment:
I just have this here because I didn't want to change your already working types. There's no need for a second `Note` type here, really.

Reply:
I'm not sure what I was talking about here.